### PR TITLE
Fix link race condition

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: router/xlink_transport/listener.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func (self *listener) bindNonSplitChannel(binding channel.Binding, routerId, rou
 
 	if existingLink, applied := self.xlinkRegistery.LinkAccepted(xli); applied {
 		log.Info("link registered")
-	} else {
+	} else if existingLink != nil {
```
Comment (dovholuknf) [inline router/xlink_transport/listener.go:227]: does it make sense to keep an actual else indicating neither match for "unexpected situations'?

File: router/handler_ctrl/dial.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func (self *dialHandler) handle(dial *ctrl_pb.Dial, _ channel.Channel) {
 				if err := self.sendLinkMessage(dial.LinkId); err != nil {
 					log.WithError(err).Error("error sending link message ")
 				}
-			} else {
+			} else if existingLink != nil {
```
Comment (dovholuknf) [inline router/handler_ctrl/dial.go:106]: same "catchall 'else' comment here"

File: controller/network/network.go [hunk 3]
```diff
@@ -330,6 +330,11 @@ func (network *Network) LinkConnected(id string, connected bool) error {
 
 	if l, found := network.linkController.get(id); found {
 		if connected {
+			if state := l.CurrentState(); state != nil && state.Mode != Pending {
+				log.Infof("link is %v not pending, cannot mark connected", state.Mode)
```
Comment (dovholuknf) [inline controller/network/network.go:334]: maybe "link **state** is %v..."? making it a bit more clear in the log?

File: controller/network/network.go [hunk 4]
```diff
@@ -330,6 +330,11 @@ func (network *Network) LinkConnected(id string, connected bool) error {
...
+				return nil
```
Comment (dovholuknf) [inline controller/network/network.go:335]: why return nil - why not return the error (and then not log)

File: router/xlink_transport/listener.go (hunk 1 above)
Comment (plorenz) [reply router/xlink_transport/listener.go:227]: The other condition (not applied, no existing link) is logged in LinkAccepted

File: controller/network/network.go (hunk 3 above)
Comment (plorenz) [reply controller/network/network.go]: fixed

File: controller/network/network.go (hunk 4 above)
Comment (plorenz) [reply controller/network/network.go]: Fixed. Removed all logging from this method, since we were logging in calling methods.

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**